### PR TITLE
Feature/filtering

### PR DESCRIPTION
**Pull request by mmeier274**
Type of change:

  [X] New feature
  
  [] Bug Fix
  
  [] Refactor
  

Short description:

  - Added show starred ideas function to only show ideas that are favorited
  - Added function to toggle show all ideas cards
  - Added toggle function to show all starred button to say show all ideas
  - Added function to search bar to filter ideas based on input value

Check the correct box:

  [X] This broke nothing
  
  [] This broke some stuff
  
  [] This broke everything
  

Checklist:

  [X] My code has no unused/commented out code
  
  [X] I have reviewed my code
  
  [X] I have fully tested my code

### CODE REVIEW

**Comment by Prkr93** [comment]
Great work on the project so far! Looks like we'll have to spend a little bit of time on Monday refactoring things and cleaning everything up, but things are looking great!

File: main.js [hunk 1]
```diff
@@ -77,8 +94,30 @@ function deleteIdea(e) {
...
+  var ideasToHide = ideas.filter(idea => !idea.title.includes(searchTextField.value) && !idea.body.includes(searchTextField.value));
+  var ideasToShow = ideas.filter(idea => idea.title.includes(searchTextField.value) || idea.body.includes(searchTextField.value));
+  for (var i = 0; i < ideasToHide.length; i++) {
+    var selector = document.getElementById(ideasToHide[i].id);
+    selector.classList.add('hidden');
+  };
+  for (var i = 0; i < ideasToShow.length; i++) {
+    var selector = document.getElementById(ideasToShow[i].id);
+    selector.classList.remove('hidden');
+  };
```
Comment (CHaiz15) [inline main.js]: I like the logic here. But I think theres a bit of "over engineering" going on. How could this be simplified? Could we do the same thing by only looping through the ideas array?

**Review by CHaiz15 — APPROVED** [review]
👍

**Comment by kmewrd** [comment]
I think that last commit helps a lot! We could possibly simplify it a bit more, but we can revisit that during our next round of refactoring. Everything else looks great. 🤙

Edit: "It" being the filterIdeas function.

**Pull request merged**